src/spikeinterface/widgets/peaks_on_probe.py

```diff
@@ -0,0 +1,170 @@
+        A SpikeInterface 'peak_locations' array created with `localize_peaks()`.
+        an array of length num_peaks with entries: (x, y)
+    segment_index : None | int, default: None
+        If set, only peaks from this recording segment will be used.
+    times_s : None | Tuple, default: None
+        The time period over which to include peaks. If `None`, peaks
+        across the entire recording will be shown.
+    ylimits : None | Tuple, default: None
+        The y-axis limits (i.e. the probe height). If `None`, the entire
+        probe will be displayed.
+    decimate : int
+        For performance reasons, every nth peak is shown on the plot,
+        where n is set by decimate. To plot all peaks, set `decimate=1`.
+    """
+
+    def __init__(
+        self,
+        recording,
+        peaks,
+        peak_locations,
+        segment_index=None,
+        times_s=None,
+        ylimits=None,
+        decimate=5,
+        backend=None,
+        **backend_kwargs,
+    ):
+        data_plot = dict(
+            recording=recording,
+            peaks=peaks,
+            peak_locations=peak_locations,
+            segment_index=segment_index,
+            times_s=times_s,
+            ylimits=ylimits,
+            decimate=decimate,
+        )
+
+        BaseWidget.__init__(self, data_plot, backend=backend, **backend_kwargs)
+
+    def plot_matplotlib(self, data_plot, **backend_kwargs):
+        import matplotlib.pyplot as plt
+        from .utils_matplotlib import make_mpl_figure
+        from spikeinterface.widgets import plot_probe_map
+
+        dp = to_attr(data_plot)
+
+        peaks, peak_locations = self._check_and_format_inputs(
+            dp.peaks,
+            dp.peak_locations,
+        )
+        fs = dp.recording.get_sampling_frequency()
+        num_plots = len(peaks)
+
+        # Set the maximum time to the end time of the longest segment
+        if dp.times_s is None:
+            seg_times = (
+                np.array([dp.recording.get_num_samples(seg) for seg in range(dp.recording.get_num_segments())]) / fs
+            )
+            times_s = (0, np.max(seg_times))
+        else:
+            times_s = dp.times_s
+
+        ## Create the figure and axes
+        if backend_kwargs["figsize"] is None:
+            backend_kwargs.update(dict(figsize=(12, 8)))
+
+        self.figure, self.axes, self.ax = make_mpl_figure(num_axes=num_plots, **backend_kwargs)
+        self.axes = self.axes[0]
+
+        # Plot each passed peaks / peak_locations over the probe on a separate subplot
+        for ax_idx, (peaks_to_plot, peak_locs_to_plot) in enumerate(zip(peaks, peak_locations)):
+
+            ax = self.axes[ax_idx]
+            plot_probe_map(dp.recording, ax=ax)
+
+            # Only keep peaks within the specified time window
+            time_mask = (peaks_to_plot["sample_index"] > int(fs * times_s[0])) & (
+                peaks_to_plot["sample_index"] < int(fs * times_s[1])
+            )
```

Comment:
If the `recording.has_time_vector`, should we use that instead?

Reply:
Good point! I added some code but will look over carefully to check. Actually, I think I should be using the `time_to_sample_index()` in this latest commit to abstract away handling of `has_time_vector`, `t_start` etc?

More generally, is there a need for the `time_vector` construct to be held at all? Could the user / us only focus on providing / handling the segments `t_start` with all time vectors constructed only the fly from `t_start`, `sampling_frequency` and segment `num_samples` when required?

Reply:
yes, I forgot we habe that function! :)

So for some systems it's important to kee a time vector, because there might be tiny misalignments in time (e.g. Neuropixels). This is important when it comes to retrieving spike times from the associated registered recording (with `sorting.get_unit_spike_train(unit_id, return_times=True)`) and correlate them with other behavioral variables on an external plot.

Reply:
Thanks @alejoe91 that makes sense!